packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/NamedTypeSymbolProvider.cs

```diff
@@ -158,6 +161,10 @@ protected override MethodProvider[] BuildMethods()
             return [.. methods];
         }
 
+        protected override bool GetIsEnum() => _namedTypeSymbol.TypeKind == TypeKind.Enum;
+
+        protected override CSharpType BuildEnumUnderlyingType() => new CSharpType(typeof(int));
```

Comment:
For custom code, the only possible underlying enum type is int, as we only consider custom code as an Enum if it uses the `enum` keyword.

Reply:
should this be `int` if it is an enum or throw if it is not an enum?